boards: linum: update the dts file and fixes build errors due to changes on some properties

uLipe opened the pull request:
Also this PR adds the possibility to push LVGL data to the SDRAM when using display subsytem present on the board.

kartben commented:
Can this be more than just 1 commit please? It's hard to believe that a "fixes a build error" commit jas to be touching so many different things, and that makes reviewing it really difficult

uLipe committed:
boards: witte: linum: update the dts file

* Add missing features for display and touch
* Add the external memories
* Fix build errors caused by enet and fdcan nodes that
have deprecated properties

Signed-off-by: Felipe Neves <ryukokki.felipe@gmail.com>

uLipe committed:
boards: witte: linum: add linker files

To place relevant display and LVGL data into
the SDRAM area. Also update the Kconfig to enable
the memory controller when the display controller is also
enabled

Signed-off-by: Felipe Neves <ryukokki.felipe@gmail.com>

uLipe commented:
@kartben, I splitted this patchset into two commits.

erwango commented on boards/witte/linum/linum.dts:
```diff
@@ -21,72 +23,64 @@
 		zephyr,canbus = &fdcan1;
+		zephyr,display = &ltdc;
+		zephyr,flash-controller = &w25q128jv;
+		zephyr,touch = &ft5446;
 	};
 
-	sdram1: sdram@c0000000 {
-		compatible = "zephyr,memory-region", "mmio-sram";
+	sdram1:sdram@c0000000 {
+		compatible = "zephyr,memory-region","mmio-sram";
```
Please move formatting changes to a dedicated commit.

uLipe replied:
Ugh it happened when I passed the file through clang-format 🤦🏻‍♂️

erwango commented on boards/witte/linum/linum.dts:
```diff
@@ -161,82 +186,116 @@
-&timers12 {
-	st,prescaler = <10000>;
+&rtc{
+	clocks = <&rcc STM32_CLOCK_BUS_APB4 0x00010000>,<&rcc STM32_SRC_LSE RTC_SEL(1)>;
```
Please use (not so) new formatting:
```suggestion
	clocks = <&rcc STM32_CLOCK(APB4, 16)>,<&rcc STM32_SRC_LSE RTC_SEL(1)>;
```

uLipe committed:
boards: witte: linum: update the dts file

To remove the hardcoded clock definitions and replace them
with the STM32H7 bindings already offered by Zephyr.

Signed-off-by: Felipe Neves <ryukokki.felipe@gmail.com>

erwango commented on boards/witte/linum/linum.dts:
```diff
@@ -135,6 +138,28 @@
+/* See reference manual (RM0433):
+ *   0b0001: division by 1
+ */
+#define MCO1_PRE_DIV_1 1
```
Not resolved

erwango commented on boards/witte/linum/linum.dts:
```diff
@@ -138,20 +139,10 @@
-	clocks = <&rcc STM32_SRC_HSE MCO1_SEL(MCO1_SEL_HSE)>;
-	prescaler = <MCO1_PRE(MCO1_PRE_DIV_1)>;
+	clocks = <&rcc STM32_SRC_HSE MCO1_SEL(STM32_SRC_HSE)>;
+	prescaler = <MCO1_PRE(MCO_PRE_DIV_1)>;
```
Don't do fixup commits, but fix in initial commit.

erwango commented on boards/witte/linum/linum.dts:
```diff
@@ -221,7 +212,7 @@
 
 &rtc{
-	clocks = <&rcc STM32_CLOCK_BUS_APB4 0x00010000>,<&rcc STM32_SRC_LSE RTC_SEL(1)>;
+	clocks = <&rcc STM32_CLOCK(APB4, 16)>,<&rcc STM32_SRC_LSE RTC_SEL(1)>;
```
Ditto

erwango commented on boards/witte/linum/linum.dts:
```diff
@@ -7,6 +7,7 @@
 /dts-v1/;
 #include <st/h7/stm32h753Xi.dtsi>
 #include <st/h7/stm32h753bitx-pinctrl.dtsi>
+#include <zephyr/dt-bindings/clock/stm32h7_clock.h>
```
Likely not required (included in .dtsi files)

erwango commented on boards/witte/linum/linum.dts:
```diff
@@ -115,11 +108,21 @@
-	div-r = <3>;
+	div-r = <10>;
+	clocks = <&clk_hse>;
+	status = "okay";
```
This clock update is not documented. It would be helpfull to split i a dedicated commit to explain this change: why it is safe or required.

erwango reviewed (changes requested):
First commit is still messy and all these changes make it difficult to understand what is done, why and eventually provide a useful review.
